File: backend-rails/app/models/group.rb
# == Schema Information
#
# Table name: groups
#
#  id         :bigint           not null, primary key
#  name       :string           not null
#  created_at :datetime         not null
#  updated_at :datetime         not null
#
class Group < ApplicationRecord
  validates :name, presence: true

  has_many :users, dependent: :nullify
  has_many :habits, dependent: :nullify
  has_many :daily_habits, dependent: :destroy

  def missing_daily_habits(user:, existing_daily_habits:)
    existing_habits = existing_daily_habits.map(&:habit)
    missing_habits = habits - existing_habits
    missing_habits.map do |habit|
      DailyHabit.build_daily_habit_from_habit(habit:, user:)
    end
  end
end

File: node/src/domain/entity/Group.ts
import { Habit } from "./Habit"
import { User } from "./User"

export class Group{
    public name!: string
    public habits: Habit[] = []
    public users: User[] = []

    constructor(name: string){
        this.name = name
    }

    addHabit(habit: Habit){
        this.habits.push(habit)
    }

    addUser(user: User){
        this.users.push(user)
    }

}